src/main/java/com/mcmoddev/communitymod/erdbeerbaerlp/iconmod/IconClassTransformer.java

```diff
@@ -0,0 +1,145 @@
+package com.mcmoddev.communitymod.erdbeerbaerlp.iconmod;
+import static org.objectweb.asm.Opcodes.*;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+import org.apache.commons.io.FileUtils;
+import org.lwjgl.opengl.Display;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.tree.*;
+import org.objectweb.asm.tree.analysis.Frame;
+
+import com.mcmoddev.communitymod.CommunityGlobals;
+import com.mcmoddev.communitymod.CommunityMod;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+import net.minecraft.util.ResourceLocation;
+import scala.tools.asm.Label;
+
+public class IconClassTransformer implements IClassTransformer{
+	private static final String[] classesBeingTransformed =
+		{
+				"net.minecraft.client.Minecraft"
+		};
+
+
+	@Override
+	public byte[] transform(String name, String transformedName, byte[] classBeingTransformed)
+	{
+		boolean isObfuscated = !name.equals(transformedName);
+		int index = Arrays.asList(classesBeingTransformed).indexOf(transformedName);
+		return index != -1 ? transform(index, classBeingTransformed, isObfuscated) : classBeingTransformed;
+	}
+
+	private static byte[] transform(int index, byte[] classBeingTransformed, boolean isObfuscated)
+	{
+		try
+		{
+			ClassNode classNode = new ClassNode();
+			ClassReader classReader = new ClassReader(classBeingTransformed);
+			classReader.accept(classNode, 0);
+			switch(index)
+			{
+
+			case 0:
+				transformMinecraft(classNode, isObfuscated);
+				transformMinecraft2(classNode, isObfuscated);
+				break;
+
+			}
+
+			ModClassWriter classWriter = new ModClassWriter(ModClassWriter.COMPUTE_MAXS | ModClassWriter.COMPUTE_FRAMES);
+			classNode.accept(classWriter);
+			return classWriter.toByteArray();
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+		return classBeingTransformed;
+	}
+	private static void transformMinecraft(ClassNode classNode, boolean isObfuscated) {
+		final String GET = isObfuscated? "func_175609_am":"createDisplay";

```

Comment:
In the future please use `FMLDeobfuscatingRemapper` to remap SRG names to MCP names. This breaks if loaded into a development environment with different mappings. It will not likely be an issue for us, just worth knowing.